socallinuxexpo/scale-network
# [READY] Ansible inventory.py python3

## Description of PR
Updates the dynamic inventory script to python3 and deduplicates the vlan dictionary.

## Previous Behavior
- inventory.py was python2
- mutable global variables made it a bit harder to read
- manually calculating IPv6 PTRs to avoid 3rd party libs and pip
- vlan dictionary was a redundant hostvar of servers

## New Behavior
- inventory.py is python3
- no mutable global variables, improves ability to follow program flow through main function
- leverage ipaddress library which is now standard in python3, no pip required
- vlan dictionary is now a variable of the "all" group, deduplicating it reduces inventory output from ~4200 lines to ~2100 and makes the vlan data available to any host or group based playbook directly
- forces ansible to ignore ssh configs, useful if you use a complex ssh config and want to test with little hassle
- dns and dhcp templates updated to reflect vlan's new location in the inventory
- adds reboot step after updating via apt
- Vagrantfile has been linted a bit to reflect modern ruby style
- Virtualbox config tweaked a bit

## Tests
1. manually check expected output with `inventory.py | jq "." | less`
2. run `vagrant up` and spot check services [ntp, dns, dhcp]
3. run `vagrant up; vagrant provision` and verify idempotent 2nd run [no changes]
4. scheduled long running test job with bash command to verify playbook logic can survive multiple executions and host removals `vagrant destroy -f; sleep 20; vagrant up && sleep 20 && vagrant provision && sleep 20 && vagrant provision && vagrant provision && vagrant destroy -f server3; vagrant up; sleep 20; vagrant provision && sleep 20 && vagrant provision && vagrant destroy -f server8; vagrant destroy -f server2; vagrant up; sleep 20 && vagrant provision && sleep 20 && vagrant provision`
5. test dns requests for `scale.lan`
6. review dhcp logs and dns query logs
7. test ntp with `chronyc`
